fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/LogicalOlapScanToPhysicalOlapScan.java

```diff
@@ -59,7 +63,17 @@ public Rule build() {
     }
 
     private DistributionSpec convertDistribution(LogicalOlapScan olapScan) {
-        DistributionInfo distributionInfo = olapScan.getTable().getDefaultDistributionInfo();
+        OlapTable olapTable = olapScan.getTable();
+        DistributionInfo distributionInfo = olapTable.getDefaultDistributionInfo();
+        ColocateTableIndex colocateTableIndex = Env.getCurrentColocateIndex();
+        if ((colocateTableIndex.isColocateTable(olapTable.getId())
+                && !colocateTableIndex.isGroupUnstable(colocateTableIndex.getGroup(olapTable.getId())))
+                || olapTable.getPartitionInfo().getType() == PartitionType.UNPARTITIONED
+                || olapTable.getPartitions().size() == 1) {
+            if (!(distributionInfo instanceof HashDistributionInfo)) {
+                return DistributionSpecAny.INSTANCE;
+            }
+        }
         if (distributionInfo instanceof HashDistributionInfo) {
```

Comment:
we should return `DistributionSpecAny.INSTANCE` even if table's `defaultDistributionInfo` is `HashDistributionInfo`, when we need to scan more than one partition and table is not colocate table or not in a stable group.

Reply:
done